pymeasure/instruments/instrument.py

```
@@ -74,12 +80,14 @@ class Instrument(CommonBase):
-            try:
+        try:
+            if isinstance(adapter, ResourceManager):
+                adapter = VISAAdapter(name, adapter, **kwargs)
+            elif isinstance(adapter, (int, str)):
                 adapter = VISAAdapter(adapter, **kwargs)
-            except ImportError:
-                raise Exception("Invalid Adapter provided for Instrument since"
-                                " PyVISA is not present")
+        except ImportError:
+            raise Exception("Invalid Adapter provided for Instrument since"
+                            " PyVISA is not present")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)